### PR TITLE
Add jemalloc static library to the static library builder for Windows & Linux

### PR DESCRIPTION
**Pull request by cculianu**
This makes the docker-based static library builder (`contrib/build/build.sh`) clone and compile jemalloc. It then links the jemalloc library statically the generated static executable.

Tested and works on both Linux and Windows.

Also in this commit: Added a "Jemalloc" key to the stats outputted by the "/stats" HTTP endpoint, which includes some of the output of `malloc_print_stats()` from Jemalloc.  Note that for this to contain any data, the jemalloc library must be linked at compile-time.

### CODE REVIEW

File: contrib/build/linux/_build.sh [hunk 1]
```diff
@@ -4,22 +4,43 @@
...
+JEMALLOC_PACKAGE="$3"
 TARGET_BINARY=Fulcrum
 
 top=/work
 cd "$top" || fail "Could not cd $top"
 . "$top/$PACKAGE/contrib/build/common/common.sh" || (echo "Cannot source common.h" && exit 1)
 
+info "Running configure for jemalloc ..."
+cd "$JEMALLOC_PACKAGE" || fail "Could not change dir to $JEMALLOC_PACKAGE"
+./autogen.sh -with-jemalloc-prefix= --disable-shared --enable-static \
```
Comment (EchterAgo) [inline contrib/build/linux/_build.sh:23]: Shouldn't it be `--with-jemalloc-prefix=` with 2 dashes?

Comment (EchterAgo) [reply contrib/build/linux/_build.sh:23]: Oh, this is for Linux, it should already default to no prefix there.

File: contrib/build/win/_build.sh [hunk 2]
```diff
@@ -4,26 +4,49 @@
...
 TARGET_ADMIN_SCRIPT=FulcrumAdmin
 
 top=/work
 cd "$top" || fail "Could not cd $top"
 . "$top/$PACKAGE/contrib/build/common/common.sh" || (echo "Cannot source common.h" && exit 1)
 
+info "Running configure for jemalloc ..."
+cd "$JEMALLOC_PACKAGE" || fail "Could not change dir to $JEMALLOC_PACKAGE"
+CXX=x86_64-w64-mingw32.static-g++ LD=x86_64-w64-mingw32.static-ld CC=x86_64-w64-mingw32.static-gcc-7.5.0 \
+    ./autogen.sh --host x86_64-w64-mingw32 -with-jemalloc-prefix= --disable-shared --enable-static \
```
Comment (EchterAgo) [inline contrib/build/win/_build.sh:25]: Same with the single dash here.

File: contrib/build/linux/_build.sh (hunk 1 above)
Comment (cculianu) [reply contrib/build/linux/_build.sh:23]: Eh -- paranoia pays!

File: contrib/build/win/_build.sh (hunk 2 above)
Comment (cculianu) [reply contrib/build/win/_build.sh:25]: Oh lol. Should be double-dash right?

Comment (cculianu) [reply contrib/build/win/_build.sh]: Yeah ok fixed.  Weird that it worked.. hmm.

File: contrib/build/linux/_build.sh (hunk 1 above)
Comment (cculianu) [reply contrib/build/linux/_build.sh]: Ok fixed. Nice catch.  Weird that it worked anyway.. maybe it accepts either?!

Comment (EchterAgo) [reply contrib/build/linux/_build.sh]: Weird if it worked, it does say `--with-jemalloc-prefix` in `configure.ac`

**Pull request merged**